### PR TITLE
Add `IndieHackers` extension (browse IH posts)

### PR DESCRIPTION
**Pull request by xmok**
## Description

This extension uses Indie Hackers RSS @ https://ihrss.io/ to fetch and display IH posts.

## Screencast

https://github.com/user-attachments/assets/0aed30f0-7bc8-4e6e-a0a7-730ba8ae31d7

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
- [x] I checked that files in the `assets` folder are used by the extension itself
- [x] I checked that assets used by the `README` are placed outside of the `metadata` folder

### CODE REVIEW

**Review by pernielsentikaer — CHANGES_REQUESTED** [review]

File: extensions/indiehackers/package.json [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+            {
+              "title": "Growth",
+              "value": "growth"
+            },
+            {
+              "title": "12 Startups in 12 Months",
+              "value": "12-startups-in-12-months"
+            }
+          ],
+          "placeholder": "POPULAR_GROUP"
```
[the remark this answers is not in the document]
Comment (pernielsentikaer) [reply extensions/indiehackers/package.json:100]: Can we look into beter placeholder names?

**Comment by xmok** [comment]
Tweaked the placeholders.
I also added a filter to make sure the items are unique because the "Today" feed seems to repeat some items which would cause duplicate key error.

**Comment by pernielsentikaer** [comment]
Should Browse by Time maybe have an optional argument instead (or even just moreve it)

What about doing the same for Browse by Group - just a dropdown in the command (And maybe a List.Section instead

I’m a bit unsure what the best thing is to do since I kinda like the arguments but they are also a bit in the way

**Comment by xmok** [comment]
Maybe we could combine all of them into one command e.g. "Browse IH" with an optional argument for group slug.
In the `List.Dropdown` we could have the Time and Type.

So if a user enters an argument, it will search by group otherwise we could set a default in Preferences and allow to toggle using the dropdown. How does that sound?

**Comment by xmok** [comment]
Here are some screenshots of how the command would look:

indiehackers-optional-slug
indiehackers-preference
indiehackers-toggle

The edge case is if a user enters a group slug, then what happens to the List.Dropdown?
option 1: we add a dynamic item `group-slug` IF there is a group slug
option 2: we hide the dropdown
option 3: we set it to an empty value and then when user selects an item from dropdown it goes back to normal

**Comment by xmok** [comment]
In this latest commit the `Time` and `Type` arguments in the commands are optional defaulting to "Today" and "Organic" respectively. For `Group` it's more difficult as there are too many groups in IH with the possibility of users creating more.

Some options:
- run some sort of query and scrape latest groups periodically showing them in a List.Dropdown
- hard code the current list of groups into a JSON file or store in a variable and update groups in future updates
- leave as is

**Comment by xmok** [comment]
I like the idea of a single command as it might be easier to browse on-the-fly. My concern is with the group browsing though maybe we could also allow changing groups by pushing a Form? I'll work on the changes and push the commit. I will not remove the previous commands for now as we can just remove the unneeded commands before merging.

**Comment by pernielsentikaer** [comment]
Sounds like a good idea. Looking forward to seeing it 🙂

**Comment by xmok** [comment]
Here's a working screencast:

https://github.com/user-attachments/assets/d42b758a-8981-4451-8043-e52c7e7d4c1c

If a group is entered, it is prepended to List.Dropdown else it's Time & Type. There is no toggle group functionality for now.

**Comment by pernielsentikaer** [comment]
Does anyone know which slugs there are? A dropdown could maybe make it easier to help users. Does that make sense?

**Comment by xmok** [comment]
It makes complete sense but it's easier said than done as there are A LOT of groups as seen @ https://www.indiehackers.com/groups e.g. there are ~20 groups in **All Active Groups** and I was able to scroll to at least 150 groups in other sections. So yes I could extract the slugs from the site and hardcode them if needed.

**Comment by pernielsentikaer** [comment]
I suppose the dropdown argument makes sense, considering that we also need to filter by date and time 😊 

There is always a tradeoff here, I think we'll do what you think is best since you know IH better than me 🙂

**Comment by xmok** [comment]
OK so for the sake of completeness, I scraped group slugs and titles from https://www.indiehackers.com/groups into a `groups.json` file. The first 20 or so are "Popular" ones. It's functioning _OK_ but I'm not sure if this is the best way to approach. Check out the new `browse-all` command to see it in action.

**Comment by pernielsentikaer** [comment]
Let's go with that 🙂

**Comment by xmok** [comment]
Renamed command to `browse-posts` and removed the other commands.

**Comment by xmok** [comment]
- Updated images ✅
- For consistency, added icons to the Type and Time items in Dropdown as well ✅

**Comment by pernielsentikaer** [comment]
✅ Approved

**Pull request merged**